wp-tevko-responsive-images.php

```diff
@@ -308,68 +307,86 @@ function tevkori_get_src_sizes( $id, $size = 'thumbnail' ) {
  * @param string $content The raw post content to be filtered.
  */
 function tevkori_filter_content_images( $content ) {
-	return preg_replace_callback(
-		'/<img\s([^>]+)>/i',
-		
-		// Don't use an anonymous callback function because it isn't supported by PHP 5.2.
-		'tevkori_filter_content_images_callback',
-		$content
-	);
+	// Only match images in our uploads directory
+	$uploads_dir = wp_upload_dir();
+	$path_to_upload_dir = $uploads_dir['baseurl'];
+
+	$content = preg_replace_callback( '|<img ([^>]+' . $path_to_upload_dir . '[^>]+) />|', '_tevkori_filter_content_images_callback', $content );
+
+	return $content;
 }
 add_filter( 'the_content', 'tevkori_filter_content_images', 5, 1 );
 
-/**
- * Callback function for tevkori_filter_content_images.
- *
- * @since 3.0
- *
- * @see tevkori_filter_content_images
- * @param array $matches Array containing the regular expression matches.
- */
-function tevkori_filter_content_images_callback( $matches ) {
-	$atts = $matches[1];
-	$sizes = $srcset = '';
-	
-	// Check if srcset attribute is not already present.
-	if ( false !== strpos( 'srcset="', $atts ) ) {
-
-		// Get the url of the original image.
-		preg_match( '/src="(.+?)(\-([0-9]+)x([0-9]+))?(\.[a-zA-Z]{3,4})"/i', $atts, $url_matches );
-		
-		$url = $url_matches[1] . $url_matches[5];
-		
-		// Get the image ID.
-		$id = attachment_url_to_postid( $url );
-
-		if ( $id ) {
-
-			// Use the width and height from the image url.
-			if ( $url_matches[3] && $url_matches[4] ) {
-				$size = array(
-					(int) $url_matches[3],
-					(int) $url_matches[4]
-				);
-			} else {
-				$size = 'full';
-			}
+function _tevkori_filter_content_images_callback( $image ) {
+	list( $image_html, $atts ) = $image;
+	$id = $size = false;
+
+	// Grab ID and size info from core classes.
+	if ( preg_match( '/wp-image-([0-9]+)/i', $atts, $class_id ) ) {
+		$id = $class_id[1];
+	}
+	if ( preg_match( '/size-([^\s|"]+)/i', $atts, $class_size ) ) {
+		$size = $class_size[1];
+	}
+
+	if ( $id && false === $size ) {
+		preg_match( '/width="([0-9]+)"/', $atts, $width );
+		preg_match( '/height="([0-9]+)"/', $atts, $height );
+
+		$size = array(
+			(int) $width[1],
+			(int) $height[1]
+		);
+	}
+
+	// If attempts to get values for `id` and `size` failed, use the
+	// src to query for matching values in `_wp_attachment_metadata`
```

Comment:
Coding standards for multi line comments https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#5-2-multi-line-comments.
Period at end of last lne.